kube-runtime/src/watcher.rs

```diff
@@ -154,6 +154,184 @@ struct FullObject<'a, K> {
     api: &'a Api<K>,
 }
 
+/// Accumulates all options that can be used on the watcher invocation.
+pub struct Config {
+    /// A selector to restrict the list of returned objects by their labels.
+    ///
+    /// Defaults to everything if `None`.
+    pub label_selector: Option<String>,
+
+    /// A selector to restrict the list of returned objects by their fields.
+    ///
+    /// Defaults to everything if `None`.
+    pub field_selector: Option<String>,
+
+    /// Timeout for the list/watch call.
+    ///
+    /// This limits the duration of the call, regardless of any activity or inactivity.
+    /// If unset for a watch call, we will use 290s.
+    /// We limit this to 295s due to [inherent watch limitations](https://github.com/kubernetes/kubernetes/issues/6513).
+    pub timeout: Option<u32>,
+
+    /// Limit the number of results.
+    ///
+    /// If there are more results, the server will respond with a continue token which can be used to fetch another page
+    /// of results. See the [Kubernetes API docs](https://kubernetes.io/docs/reference/using-api/api-concepts/#retrieving-large-results-sets-in-chunks)
+    /// for pagination details.
+    pub limit: Option<u32>,
+
+    /// Fetch a second page of results.
+    ///
+    /// After listing results with a limit, a continue token can be used to fetch another page of results.
+    pub continue_token: Option<String>,
+
+    /// Sets a constraint on what resource versions a request may be served from.
+    /// See https://kubernetes.io/docs/reference/using-api/api-concepts/#resource-versions for
+    /// details.
+    pub resource_version: Option<String>,
+
+    /// Determines how resourceVersion is applied to list calls.
+    /// It is highly recommended that resourceVersionMatch be set for list calls where
+    /// resourceVersion is set
+    /// See https://kubernetes.io/docs/reference/using-api/api-concepts/#resource-versions for
+    /// details.
+    pub resource_version_match: Option<ResourceVersionMatch>,
+
+    /// Enables watch events with type "BOOKMARK".
+    ///
+    /// Servers that do not implement bookmarks ignore this flag and
+    /// bookmarks are sent at the server's discretion. Clients should not
+    /// assume bookmarks are returned at any specific interval, nor may they
+    /// assume the server will send any BOOKMARK event during a session.
+    /// If this is not a watch, this field is ignored.
+    /// If the feature gate WatchBookmarks is not enabled in apiserver,
+    /// this field is ignored.
+    pub bookmarks: bool,
+}
+
+impl From<&Config> for ListParams {
+    fn from(val: &Config) -> ListParams {
+        ListParams {
+            label_selector: val.label_selector.clone(),
+            field_selector: val.field_selector.clone(),
+            timeout: val.timeout,
+            limit: val.limit,
+            continue_token: val.continue_token.clone(),
```

Comment:
we should do these via private converters for now i think. mostly because we shouldn't have limit + continue_token in Config

Reply:
Added two private converters